### PR TITLE
Implement a Double basis module

### LINKED ISSUE #637: Implement a `Double` basis module
The `Double` basis module should be implemented once #636 has been resolved.
The goal should be to define a basis module `Double` which provides functions for operations over double, including `fromString` and `toString` functions; the double type should be abstract

### PR DESCRIPTION
**Pull request by HeikoBecker**
This implements a basis module for double floating-point arithmetic (fixes #637 and #638)
Parsing of floating-point words is currently done over the FFI using functions `fromString` and `toString` in `basis_ffi.c`.
File `DoubleProofsScript.sml` proves that the from and toString functions behave correctly under the assumption that the FFI is implemented correctly.

### CODE REVIEW

File: basis/DoubleFFIScript.sml [hunk 1]
```diff
@@ -0,0 +1,77 @@
+(*
+  Logical model of the Double state: TODO
```
Comment (myreen) [inline basis/DoubleFFIScript.sml]: This TODO should be removed before this gets merged.

Comment (HeikoBecker) [reply basis/DoubleFFIScript.sml]: Done.

File: basis/DoubleProgScript.sml [hunk 2]
```diff
@@ -0,0 +1,123 @@
+(*
+  Module about the built-in integer tyoe. Note that CakeML uses
```
Comment (myreen) [inline basis/DoubleProgScript.sml]: This comment doesn't seem right. The same comment appears in `README.md`.

File: basis/README.md [hunk 3]
```diff
@@ -17,6 +17,16 @@ library.
 [CommandLineProofScript.sml](CommandLineProofScript.sml):
 Proof about the command-line module of the CakeML standard basis library.
 
+[DoubleFFIScript.sml](DoubleFFIScript.sml):
+Specifies behaviour of FFI calls from/toString for doubles
+
+[DoubleProgScript.sml](DoubleProgScript.sml):
```
Comment (myreen) [inline basis/README.md]: Update this by running `Holmake --no_prereqs README.md` in `basis` after improving the comment at the top of the `DoubleFFIScript.sml` file.

**Pull request merged**